systems-state-manager/nisystemsstate.yml

```diff
@@ -0,0 +1,999 @@
+  /v1/replace-state-content:
+    post:
+      operationId: ReplaceContent
+      x-ni-privilege: nisystemsstate.Write
+      x-ni-operation: createOrUpdateStates
+      tags: [states]
+      summary: Replace the content of a state
+      description: Replace the content of a state by providing the new content of the state.
+      consumes:
+        - multipart/form-data
+      parameters:
+        - name: id
+          in: formData
+          description: ID of the state for which to update the content.
+          required: true
+          type: string
+        - in: formData
+          description: New content of the state.
+          name: file
+          type: file
+          required: true
+        - in: formData
+          description: Description of the change operation.
+          name: changeDescription
+          type: string
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: '#/definitions/StateResponse'
+        400:
+          $ref: '#/responses/BadRequest'
+        401:
+          $ref: '#/responses/Unauthorized'
+        409:
+          $ref: '#/responses/Conflict'
+        default:
+          $ref: '#/responses/Error'
+  /v1/states/{id}/history:
+    get:
+      operationId: GetStateHistory
+      x-ni-privilege: Read
+      x-ni-operation: getStates
+      tags: [history]
+      summary: Get state history
+      description: Returns a list of (version, description) pairs defining the history of the state.
+      parameters:
+        - name: id
+          in: path
+          description: ID of the state for which it returns the history.
+          required: true
+          type: string
+        - in: query
+          name: skip
+          description: How many state versions to skip. Used for pagination.
+          type: integer
+          x-ni-data-type: U32
+          default: 0
+        - in: query
+          name: take
+          description: How many state versions to retrieve. Used for pagination.
+          type: integer
+          x-ni-data-type: U32
+          default: 1000
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: '#/definitions/StateHistoryResponse'
+        401:
+          $ref: '#/responses/Unauthorized'          
+        404:
+          $ref: '#/responses/NotFound'
+        default:
+          $ref: '#/responses/Error'
+  /v1/states/{id}/history/{version}:
+    get:
+      operationId: GetStateVersion
+      x-ni-privilege: Read
+      x-ni-operation: getStates
+      tags: [history]
+      summary: Get state by version
+      description: Returns the snapshot of a state at a specific version. Returns the metadata, the feeds and packages from the content of the state. If the state contains only feeds and packages entries the containsExtraOperations from response will be false, otherwise it will be true. The export route could be used in order to get the whole content of the state.
```

Comment:
"Returns the snapshot of a state for a specific version." Then please reuse my suggested description (if you think it's a good suggestion, that is) from line 651.